test/e2e_flags.go

```diff
@@ -64,11 +64,17 @@ func initializeFlags() *EnvironmentFlags {
 	flag.StringVar(&f.Cluster, "cluster", "",
 		"Provide the cluster to test against. Defaults to the current cluster in kubeconfig.")
 
-	var defaultKubeconfig string
-	if usr, err := user.Current(); err == nil {
-		defaultKubeconfig = path.Join(usr.HomeDir, ".kube/config")
+	// Use KUBECONFIG if available
+	defaultKubeconfig := os.Getenv("KUBECONFIG")
+
+	// If KUBECONFIG env var isn't set then look for $HOME/.kube/config
+	if defaultKubeconfig == "" {
+		if usr, err := user.Current(); err == nil {
+			defaultKubeconfig = path.Join(usr.HomeDir, ".kube/config")
+		}
 	}
 
```

Comment:
nit, felt a log here would be helpful for debugging purpose

Reply:
ok done

Reply:
Actually I didn't realize there was a flag override before seeing your change. I'm not quite convinced that the flag was parsed at the point of logging, so it might not even be doable here. Sorry for misleading